### ISSUE TITLE
_pickle.UnpicklingError: invalid load key

### ISSUE THREAD

**Issue by ioctl-user** [Srameo/LED#18]
I was running clean ubuntu 20.04 in docker. After installing gcc, make, conda, etc.. I have downloaded LED repo following README.md instructions and run install.sh. After that I was downloaded pth-model, activated conda environment and run test script, which is failed:

```
(LED-ICCV23) root@host:/ICCV23-LED# python3 /ICCV23-LED/scripts/benckmark.py -p /ICCV23-LED/pretrained/SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth --dataset SID SonyA7S2 -t test
Disable distributed.
Path already exists. Rename it to /ICCV23-LED/results/test_archived_20231213_115820
2023-12-13 11:58:20,049 INFO:
     ______                   __   __                 __      __
    / ____/____   ____   ____/ /  / /   __  __ _____ / /__   / /
   / / __ / __ \ / __ \ / __  /  / /   / / / // ___// //_/  / /
  / /_/ // /_/ // /_/ // /_/ /  / /___/ /_/ // /__ / /<    /_/
  \____/ \____/ \____/ \____/  /_____/\____/ \___//_/|_|  (_)

Version Information:
        LED: 0.0.1
        PyTorch: 1.13.1+cu117
        TorchVision: 0.14.1+cu117
2023-12-13 11:58:20,049 INFO:
  val:[
    val_freq: 1288.0
    save_img: False
    suffix: None
    calculate_metric_in_batch: True
    illumination_correct: True
    metric_in_srgb: False
    metrics:[
      psnr:[
        type: calculate_psnr
        crop_border: 2
        test_y_channel: False
      ]
      ssim:[
        type: calculate_ssim
        crop_border: 2
        test_y_channel: False
      ]
    ]
  ]
  logger:[
    print_freq: 200
    save_checkpoint_freq: 1288.0
    use_tb_logger: True
    wandb: None
  ]
  datasets:[
    val_100:[
      name: SIDSonyA7S2Paired100
      type: PairedRAWDataset
      dataroot: datasets/ICCV23-LED/Sony_npy
      postfix: npz
      which_meta: gt
      data_pair_list: datasets/txtfiles/SID/SonyA7S2/valid_100.txt
      zero_clip: False
      crop_size: None
      load_in_mem: False
      phase: val
      scale: 1
    ]
    val_250:[
      name: SIDSonyA7S2Paired250
      type: PairedRAWDataset
      dataroot: datasets/ICCV23-LED/Sony_npy
      postfix: npz
      which_meta: gt
      data_pair_list: datasets/txtfiles/SID/SonyA7S2/valid_250.txt
      zero_clip: False
      crop_size: None
      load_in_mem: False
      phase: val
      scale: 1
    ]
    val_300:[
      name: SIDSonyA7S2Paired300
      type: PairedRAWDataset
      dataroot: datasets/ICCV23-LED/Sony_npy
      postfix: npz
      which_meta: gt
      data_pair_list: datasets/txtfiles/SID/SonyA7S2/valid_300.txt
      zero_clip: False
      crop_size: None
      load_in_mem: False
      phase: val
      scale: 1
    ]
  ]
  network_g:[
    type: UNetArch
    inchannels: 4
    outchannels: 4
    channels: 32
  ]
  base: ['options/base/network_g/unet.yaml', 'options/base/dataset/test/SID_SonyA7S2_val_split.yaml', 'options/base/val_and_logger.yaml']
  name: test
  model_type: RAWImageDenoisingModel
  scale: 1
  num_gpu: 1
  manual_seed: 2022
  path:[
    pretrain_network_g: /ICCV23-LED/pretrained/SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth
    strict_load_g: True
    resume_state: None
    CRF: datasets/EMoR
    results_root: /ICCV23-LED/results/test
    log: /ICCV23-LED/results/test
    visualization: /ICCV23-LED/results/test/visualization
  ]
  dist: False
  rank: 0
  world_size: 1
  auto_resume: False
  is_train: False

2023-12-13 11:58:20,049 INFO: Dataset [PairedRAWDataset] - SIDSonyA7S2Paired100 is built.
2023-12-13 11:58:20,050 INFO: Number of test images in SIDSonyA7S2Paired100: 29
2023-12-13 11:58:20,050 INFO: Dataset [PairedRAWDataset] - SIDSonyA7S2Paired250 is built.
2023-12-13 11:58:20,050 INFO: Number of test images in SIDSonyA7S2Paired250: 29
2023-12-13 11:58:20,051 INFO: Dataset [PairedRAWDataset] - SIDSonyA7S2Paired300 is built.
2023-12-13 11:58:20,051 INFO: Number of test images in SIDSonyA7S2Paired300: 35
2023-12-13 11:58:20,143 INFO: Network [UNetArch] is created.
2023-12-13 11:58:21,088 INFO: Network: UNetArch, with parameters: 7,760,484
2023-12-13 11:58:21,088 INFO: UNetArch(
  (conv1_1): Conv2d(4, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv1_2): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (pool1): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  (conv2_1): Conv2d(32, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv2_2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (pool2): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  (conv3_1): Conv2d(64, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv3_2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (pool3): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  (conv4_1): Conv2d(128, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv4_2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (pool4): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  (conv5_1): Conv2d(256, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv5_2): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (upv6): ConvTranspose2d(512, 256, kernel_size=(2, 2), stride=(2, 2))
  (conv6_1): Conv2d(512, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv6_2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (upv7): ConvTranspose2d(256, 128, kernel_size=(2, 2), stride=(2, 2))
  (conv7_1): Conv2d(256, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv7_2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (upv8): ConvTranspose2d(128, 64, kernel_size=(2, 2), stride=(2, 2))
  (conv8_1): Conv2d(128, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv8_2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (upv9): ConvTranspose2d(64, 32, kernel_size=(2, 2), stride=(2, 2))
  (conv9_1): Conv2d(64, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv9_2): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1))
  (conv10_1): Conv2d(32, 4, kernel_size=(1, 1), stride=(1, 1))
)
Traceback (most recent call last):
  File "led/test.py", line 45, in <module>
    test_pipeline(root_path)
  File "led/test.py", line 35, in test_pipeline
    model = build_model(opt)
  File "/ICCV23-LED/led/models/__init__.py", line 26, in build_model
    model = MODEL_REGISTRY.get(opt['model_type'])(opt)
  File "/ICCV23-LED/led/models/raw_image_denoising_model.py", line 56, in __init__
    self.load_network(self.net_g, load_path, self.opt['path'].get('strict_load_g', True), param_key)
  File "/ICCV23-LED/led/models/base_model.py", line 305, in load_network
    load_net = torch.load(load_path, map_location=lambda storage, loc: storage)
  File "/opt/conda/envs/LED-ICCV23/lib/python3.8/site-packages/torch/serialization.py", line 795, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/conda/envs/LED-ICCV23/lib/python3.8/site-packages/torch/serialization.py", line 1002, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
_pickle.UnpicklingError: invalid load key, '<'.
```

How to fix this?

**Comment by Srameo** [comment]
Hi there, could you please check the MD5 checksum of file `SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth`? It's possible that an issue with this .pth file is causing the read failure. The correct checksum should be `c8809c84a88bb7eae5548866c0b9c5d5`.

**Comment by ioctl-user** [comment]
Something wrong with the download script:
```
(LED-ICCV23) root@host:/ICCV23-LED# python3 scripts/download_gdrive.py --id 1Rz5oShriEBFYQLiIXL8uKVhkWFsZuY9P --save-path /SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth
(LED-ICCV23) root@host:/ICCV23-LED# echo $?
0
(LED-ICCV23) root@host:/ICCV23-LED# md5sum /SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth
2678377344be7429a4d879ce79ca8de6  /SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth
(LED-ICCV23) root@host:/ICCV23-LED# ls -lh /SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth
-rw-r--r-- 1 root root 2.3K Dec 13 15:49 /SID_Deploy_SID_SonyA7S2_CVPR20_Setting_Ratio100-300.pth
```

File content looks like this:
![pth](https://github.com/Srameo/LED/assets/23457638/9c881825-6add-4953-9969-385a6491ff83)

So, it's not file itself, it's Google warning before download.

And I just find script download_all_network_g.sh . It has the same problem.

**Comment by Srameo** [comment]
We apologize for the issues you have encountered with the use of LED and thank you for your suggestions. 

We will determine a solution soon and re-upload the relevant files!

**Comment by Srameo** [comment]
We have released our checkpoint in the [release v0.1.0](https://github.com/Srameo/LED/releases/tag/0.1.0).

**Comment by ioctl-user** [comment]
Thanks!

Seems, issue is over.
